src/index.js

```diff
@@ -183,7 +183,8 @@ const userEvent = {
     };
     const opts = Object.assign(defaultOpts, userOpts);
 
-    const computedText = text.slice(0, element.maxLength || text.length);
+    const maxLength = element.maxLength > 0 ? element.maxLength : text.length;
```

Comment:
From the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength):

> The maxlength attribute defines the maximum number of characters (as UTF-16 code units) the user can enter into an <input> or <textarea>. This must be an integer value 0 or higher. If no maxlength is specified, or an invalid value is specified, the input or textarea has no maximum length.

So it seems like we wouldn't want to check for `maxLength > 0`, because `0` is a valid value for maxLength. It needs to check for `>= 0` instead, which necessitates another test for `maxLength: 0`

@a-bates